apps/zotonic_core/src/models/m_rsc_update.erl

```diff
@@ -657,6 +657,18 @@ update_result({ok, NewId, {OldProps, NewProps, OldCatList, IsCatInsert}}, #rscup
         },
         Context),
 
+    % If a newly inserted resource is dependent, then schedule a check
+    % if it is connected. If not then it should be deleted.
+    case maps:get(<<"is_dependent">>, NewProps, false) of
+        true ->
+            case maps:get(<<"is_dependent">>, OldProps, false) of
+                true -> ok;
+                false -> z_edge_log_server:maybe_schedule_dependent_check(NewId, Context)
+            end;
+        false ->
+            ok
+    end,
```

Comment:
Why not directly:
```suggestion
    z_edge_log_server:maybe_schedule_dependent_check(NewId, Context),
```
The check of the `is_dependent` value is already done in `maybe_schedule_dependent_check` and if this is called multiple times per resources (in the case of multiple updates), the task is keyed so it's not duplicated.

Reply:
It is an optimization, only run the check if the `is_dependent` has been changed.